api/src/main/java/io/smallrye/concurrency/api/ManagedExecutorConfig.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2018,2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.concurrency.api;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ThreadContext;
+
+import javax.enterprise.util.AnnotationLiteral;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Annotates a CDI injection point for a {@link ManagedExecutor} such that the container
+ * creates a new instance, which is identified within an application by its unique name.
+ * The unique name is generated as the fully qualified class name (with each component delimited by <code>.</code>)
+ * and the injection point's field name or method name and parameter position, all delimited by <code>/</code>,
+ * unless annotated with the {@link NamedInstance} qualifier,
+ * in which case the unique name is specified by the {@link NamedInstance#value} attribute of that qualifier.</p>
+ *
+ * <p>For example, the following injection points share a single
+ * {@link ManagedExecutor} instance,</p>
+ *
+ * <pre><code> {@literal @}Inject {@literal @}NamedInstance("exec1") {@literal @}ManagedExecutorConfig(maxAsync=5)
+ * ManagedExecutor executor;
+ *
+ * {@literal @}Inject
+ * void setCompletableFuture({@literal @}NamedInstance("exec1") ManagedExecutor exec) {
+ *     completableFuture = exec.newIncompleteFuture();
+ * }
+ *
+ * {@literal @}Inject
+ * void setCompletionStage({@literal @}NamedInstance("exec1") ManagedExecutor exec) {
+ *     completionStage = exec.supplyAsync(supplier);
```

Comment:
It comes from thin air ;-)
It's not defined in the sample because I didn't think it important, the focus was on the `ManagedExecutor` being a shared instance. We can improve that and toss lambda inside.